aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1204,9 +1207,16 @@ Tensor& logsumexp_out(const Tensor& self, IntArrayRef dims, bool keepdim, Tensor
 }
 
 Tensor logsumexp(const Tensor& self, IntArrayRef dims, bool keepdim) {
-  Tensor result = at::empty({0}, self.options());
-  return at::native::logsumexp_out(self, dims, keepdim, result);
+  Tensor result;
+  if (at::isIntegralType(self.scalar_type(), /*includeBool=*/true)) {
+    result = at::empty({0}, self.options().dtype(kFloat));
+    return at::native::logsumexp_out(self.to(kFloat), dims, keepdim, result);
```

Comment:
is it necessary to cast the self tensor to float as well? Could you try if it would work without this cast?

Reply:
Sorry for the delay, without the casting of `self` tensor , It would raise `RuntimeError: value cannot be converted to type int16_t without overflow: inf`. That caused by the comparion between integral `maxes_squeezed` and `INFINITY` here. ↓ 
https://github.com/pytorch/pytorch/blob/d6133b2fe6b863dd49fb21641bd04e24e19ac794/aten/src/ATen/native/ReduceOps.cpp#L1185-L1187
Even comment out above comparion, will also meet `RuntimeError: result type Float can't be cast to the desired output type Long`. So I think the promote of integral inputs might be necessary, and the dtype of `result` tensor should keep consistent with `self` as well. Please correct me, If I've missed something. 💦

Reply:
Sounds good! thanks for verifying this.